sbtc-core/src/signer/coordinator/fire.rs

```diff
@@ -1,2 +1,28 @@
 // TODO: FIRE coordination logic
 // https://github.com/Trust-Machines/stacks-sbtc/issues/667
+
+use crate::signer::{
+    coordinator::{Coordinate, PublicKeys},
+    SBTCResult,
+};
+use bitcoin::{PublicKey, Transaction as BitcoinTransaction};
+use wsts::{bip340::SchnorrProof, common::Signature};
+/// FIRE coordinator
+
+#[derive(Default)]
+pub struct Coordinator {}
+
+impl Coordinate for Coordinator {
+    /// Generate the sBTC wallet public key
+    fn generate_sbtc_wallet_public_key(&self, _public_keys: &PublicKeys) -> SBTCResult<PublicKey> {
+        todo!()
+    }
```

Comment:
Since we know this will abstract away a lot of IO and message passing, we should either split this up into separate functions like "initiate_dkg" and "get_sbtc_wallet_public_key". Optionally I'd be fine with making this async directly.

Reply:
generate_sbtc_wallet_public_key is abstracted away. If youw anted the address in the smart contract, you would go to the broker. As for communication via stacker db, the coordinator would need to read internal store that the broker would be in charge of updating I imagine.